.github/workflows/auto-update.js

```diff
@@ -1,41 +1,63 @@
-const today = new Date();
-const year = today.getFullYear();
-const month = (today.getMonth() + 1).toString().padStart(2, "0");
-const day = today.getDate().toString().padStart(2, "0");
+const { url } = require("inspector");
 
 const source_regex =
   /name = "nixos-unstable-(20[0-9\-]+)";.    url = https:\/\/github\.com\/nixos\/nixpkgs\/archive\/([0-9a-z]+)\.tar\.gz;.    sha256 = "([0-9a-z]+)";/s;
 
 module.exports = async ({ github, context, core, require }) => {
+  console.log("running...");
   const https = require("https");
+  const { URL } = require("url");
   const { readFileSync, writeFileSync } = require("fs");
   const { exec } = require("child_process");
 
-  const source_path = "./sources.nix";
-
-  const revisions = new Promise((resolve, reject) => {
-    https
-      .get(
-        "https://monitoring.nixos.org/prometheus/api/v1/query?query=channel_revision",
-        (res) => {
-          const response_data = [];
-          res.on("data", (data) => {
-            response_data.push(data);
-          });
-
-          res.on("end", () => {
-            const response = Buffer.concat(response_data);
-            const response_string = response.toString();
-            const revisions = JSON.parse(response_string).data.result;
-            resolve(revisions);
-          });
-        }
-      )
-      .on("error", reject);
-  }).then((rev) => {
-    const revision = rev.find((d) => d.metric.channel === "nixos-unstable");
-    return revision;
-  });
+  const http_request = (uri) => {
+    return new Promise((resolve, reject) => {
+      console.log("http request to: ", uri);
+      const url = new URL(uri);
+
+      console.log(url);
+
+      return https
+        .get(
+          {
+            port: 443,
+            path: url.pathname + url.search,
+            protocol: url.protocol,
+            hostname: url.hostname,
+            method: "GET",
+            headers: {
+              "User-Agent": "GithubActions",
+              Accept: "application/json",
+            },
+            url,
+          },
+          (res) => {
+            const response_data = [];
+            res.on("data", (data) => {
+              response_data.push(data);
+            });
+
+            res.on("end", () => {
+              console.log("is it ever here?");
```

Comment:
print

Reply:
Looks like I forgot to remove some debugging stuff 😂